packages/menu/src/Menu/HighlightReducer/HighlightReducer.ts

```diff
@@ -0,0 +1,40 @@
+import { type Dispatch, type Reducer, useReducer } from 'react';
+
+import { DescendantsList } from '@leafygreen-ui/descendants';
+
+import { getUpdatedIndex } from './utils/getUpdatedIndex';
+import { isDescendantsSet } from './utils/isDescendantsSet';
+import type { Direction, Index } from './highlight';
+
+const getInitialIndex = (descendants: DescendantsList<HTMLElement>) =>
+  isDescendantsSet(descendants) ? 0 : undefined;
+
+/**
+ * Custom hook that handles setting the highlighted descendant index,
+ * and fires any `onChange` side effects
+ */
+export const useHighlightReducer = (
```

Comment:
this is a cool api! I see what you were previously saying about sharing similar logic across multiple descendants use cases